BMM.Core/Implementations/DeepLinking/DeepLinkHandler.cs

```diff
@@ -121,6 +121,14 @@ private async Task OpenGenericDocumentsView(GenericDocumentsViewParameters gener
         {
             if (!_viewPresenter.IsViewModelShown<T>())
                 await _navigationService.Navigate<T>();
+
+            SendEventIfNeeded(typeof(T));
+        }
+
+        private void SendEventIfNeeded(Type type)
+        {
+            if (type == typeof(YearInReviewViewModel))
```

Comment:
Isn't it better to send this event in YearInReviewViewModel?

Reply:
Hmm, but in which place, Initialize? Constructor? I think creating a VM is not comparable to navigating to it, because it may be recreated by the operation system without intentional navigation from the user.

Reply:
sounds good